### ISSUE TITLE
[2.9] Call to undefined function mcrypt_module_get_supported_key_sizes()

### ISSUE THREAD

**Issue by jamiewade** [craftcms/cms#5602]
### Description
When I try and access the email settings area of _all_ our Craft 2 sites, I get this error:

> Call to undefined function mcrypt_module_get_supported_key_sizes() in /sites/*/craft/app/framework/base/CSecurityManager.php:535

We were getting [this error](https://github.com/craftcms/cms/issues/5417) originally, but after @angrybrad's suggestion of replacing the whole `/app` folder, we now get the above.

I get this error both on our production servers and my local sites.

### Additional info

- Craft version: Craft CMS 2.9.0
- PHP version: 7.3.11

**Comment by brandonkelly** [comment]
Can you find that error in `craft/storage/runtime/logs/craft.log(.x)` and post the stack trace leading up to it?

**Comment by jamiewade** [comment]
Here you go!

```
2020/02/06 14:04:02 [error] [exception.Error] Error: Call to undefined function mcrypt_module_get_supported_key_sizes() in C:\sites\glut\craft\app\framework\base\CSecurityManager.php:535
Stack trace:
#0 C:\sites\glut\craft\app\framework\base\CSecurityManager.php(185): CSecurityManager->validateEncryptionKey('FwSJ2zBj_OFay_y...')
#1 C:\sites\glut\craft\app\framework\base\CSecurityManager.php(172): CSecurityManager->setEncryptionKey('FwSJ2zBj_OFay_y...')
#2 C:\sites\glut\craft\app\framework\base\CSecurityManager.php(221): CSecurityManager->getEncryptionKey()
#3 C:\sites\glut\craft\app\helpers\StringHelper.php(738): CSecurityManager->encrypt('50Dickens2017')
#4 C:\sites\glut\craft\app\etc\templating\twigextensions\CraftTwigExtension.php(140): Craft\StringHelper::encenc('50Dickens2017')
#5 C:\sites\glut\craft\storage\runtime\compiled_templates\b2\b28781aa253f8d97ad0d51cf4afd624df6c2af1b0da755bc297a8acdde38ff54.php(130): Craft\CraftTwigExtension->encencFilter('50Dickens2017')
#6 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(217): __TwigTemplate_b30411de54b625995bb7df9aa5512a7bd8c46585b96fb24ca310e60cd3b530bd->block_content(Array, Array)
#7 C:\sites\glut\craft\storage\runtime\compiled_templates\37\376bf1564955aa4083890492de2a12e0c672cacee73e8f799f369a015148e8d8.php(751): Twig\Template->displayBlock('content', Array, Array)
#8 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(217): __TwigTemplate_18b928c4d0f679d034d8c1529767ddf0fbe8adc92addbc9b82e3e60d24da413d->block_body(Array, Array)
#9 C:\sites\glut\craft\storage\runtime\compiled_templates\c2\c207beca4839ad2554d971119ec4808c9c4c863d964d8ecbf9457358d519a9ec.php(56): Twig\Template->displayBlock('body', Array, Array)
#10 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(456): __TwigTemplate_2d5f72cc7451af1544620cd96313444972604b40a7e3c75bdb76082fab6a8591->doDisplay(Array, Array)
#11 C:\sites\glut\craft\app\etc\templating\BaseTemplate.php(26): Twig\Template->displayWithErrorHandling(Array, Array)
#12 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(423): Craft\BaseTemplate->displayWithErrorHandling(Array, Array)
#13 C:\sites\glut\craft\storage\runtime\compiled_templates\e9\e9ea81678bc516ba10bd70cdf5fa3370625b851e07505399511180ebc43094a9.php(145): Twig\Template->display(Array, Array)
#14 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(456): __TwigTemplate_8f36870b8a987642f5fe43aaa970a4d379406f31e4eb32798e65340fd2bc23ec->doDisplay(Array, Array)
#15 C:\sites\glut\craft\app\etc\templating\BaseTemplate.php(26): Twig\Template->displayWithErrorHandling(Array, Array)
#16 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(423): Craft\BaseTemplate->displayWithErrorHandling(Array, Array)
#17 C:\sites\glut\craft\storage\runtime\compiled_templates\37\376bf1564955aa4083890492de2a12e0c672cacee73e8f799f369a015148e8d8.php(180): Twig\Template->display(Array, Array)
#18 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(456): __TwigTemplate_18b928c4d0f679d034d8c1529767ddf0fbe8adc92addbc9b82e3e60d24da413d->doDisplay(Array, Array)
#19 C:\sites\glut\craft\app\etc\templating\BaseTemplate.php(26): Twig\Template->displayWithErrorHandling(Array, Array)
#20 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(423): Craft\BaseTemplate->displayWithErrorHandling(Array, Array)
#21 C:\sites\glut\craft\storage\runtime\compiled_templates\83\83e61a4b0b37bb7c9ab8fb0a5e70f005c4c4a20706f4ec2928a913e9fcb7531e.php(49): Twig\Template->display(Array, Array)
#22 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(456): __TwigTemplate_ed155aaea2b7b400ec8abae78e0ffd7c1ff50572653a246cdca336a57f5a6efe->doDisplay(Array, Array)
#23 C:\sites\glut\craft\app\etc\templating\BaseTemplate.php(26): Twig\Template->displayWithErrorHandling(Array, Array)
#24 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(423): Craft\BaseTemplate->displayWithErrorHandling(Array, Array)
#25 C:\sites\glut\craft\storage\runtime\compiled_templates\b2\b28781aa253f8d97ad0d51cf4afd624df6c2af1b0da755bc297a8acdde38ff54.php(69): Twig\Template->display(Array, Array)
#26 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(456): __TwigTemplate_b30411de54b625995bb7df9aa5512a7bd8c46585b96fb24ca310e60cd3b530bd->doDisplay(Array, Array)
#27 C:\sites\glut\craft\app\etc\templating\BaseTemplate.php(26): Twig\Template->displayWithErrorHandling(Array, Array)
#28 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(423): Craft\BaseTemplate->displayWithErrorHandling(Array, Array)
#29 C:\sites\glut\craft\app\vendor\twig\twig\src\Template.php(435): Twig\Template->display(Array)
#30 C:\sites\glut\craft\app\vendor\twig\twig\src\TemplateWrapper.php(47): Twig\Template->render(Array, Array)
#31 C:\sites\glut\craft\app\vendor\twig\twig\src\Environment.php(384): Twig\TemplateWrapper->render(Array)
#32 C:\sites\glut\craft\app\services\TemplatesService.php(256): Twig\Environment->render('settings/email', Array)
#33 C:\sites\glut\craft\app\controllers\BaseController.php(74): Craft\TemplatesService->render('settings/email', Array)
#34 C:\sites\glut\craft\app\controllers\TemplatesController.php(68): Craft\BaseController->renderTemplate('settings/email', Array)
#35 [internal function]: Craft\TemplatesController->actionRender('settings/email', Array)
#36 C:\sites\glut\craft\app\framework\web\actions\CAction.php(109): ReflectionMethod->invokeArgs(Object(Craft\TemplatesController), Array)
#37 C:\sites\glut\craft\app\framework\web\actions\CInlineAction.php(47): CAction->runWithParamsInternal(Object(Craft\TemplatesController), Object(ReflectionMethod), Array)
#38 C:\sites\glut\craft\app\framework\web\CController.php(308): CInlineAction->runWithParams(Array)
#39 C:\sites\glut\craft\app\framework\web\CController.php(286): CController->runAction(Object(CInlineAction))
#40 C:\sites\glut\craft\app\framework\web\CController.php(265): CController->runActionWithFilters(Object(CInlineAction), Array)
#41 C:\sites\glut\craft\app\framework\web\CWebApplication.php(282): CController->run('render')
#42 C:\sites\glut\craft\app\framework\web\CWebApplication.php(141): CWebApplication->runController('templates/rende...')
#43 C:\sites\glut\craft\app\etc\web\WebApp.php(291): CWebApplication->processRequest()
#44 C:\sites\glut\craft\app\framework\base\CApplication.php(185): Craft\WebApp->processRequest()
#45 C:\sites\glut\craft\app\index.php(73): CApplication->run()
#46 C:\sites\glut\public\index.php(21): require_once('C:\\sites\\glut\\c...')
#47 {main}
REQUEST_URI=/admin/settings/email
HTTP_REFERER=http://glut.local.jw/admin/settings
```

**Comment by brandonkelly** [comment]
Whoops, looks like thy `mcrypt_compat` library was accidentally removed in 0c9b65ee1fdc030beb966cac95e2baf6970459fa. I just released Craft 2.9.1 with it back in place.

**Comment by jamiewade** [comment]
Thanks Brandon, that's all back now 👍
